tensorboard/webapp/widgets/linked_time_fob/types.ts

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+import {ElementRef} from '@angular/core';
+
+export enum AxisDirection {
+  HORIZONTAL,
+  VERTICAL,
+}
+
+export enum Fob {
+  NONE,
+  START,
+  END,
+}
+
+export interface FobCardAdapter {
+  upperBound: number;
```

Comment:
This (upperBound and lowerBound) is internal state that you expect each implementation to have. It suggests that the state should be moved to LinkedTimeFobControllerComponent. So, in this particular case, upperBound and lowerBound would be moved back to LinkedTimeFobControllerComponent. (Aside: upperBound and lowerBound are pretty ambiguous names in this case - I think in LinkedTimeFobControllerComponent these should be named something like upperDraggableBound and lowerDraggableBound?)

What might this mean for the FobCardAdapter API? A possibility:  Instead of setBound() you could have `getCardLowerBound(): number` and `getCardUpperBound(): number`. For other API calls you could pass an argument draggableBounds: [number, number].